src/Microsoft.AspNet.Hosting/Internal/HostingApplication.cs

```diff
@@ -46,28 +46,28 @@ public Context CreateContext(IFeatureCollection contextFeatures)
             {
                 HttpContext = httpContext,
                 Scope = scope,
-                StartTick = startTick,
+                StartTimestamp = startTick,
             };
         }
 
         public void DisposeContext(Context context, Exception exception)
         {
             var httpContext = context.HttpContext;
-            var currentTick = Environment.TickCount;
-            _logger.RequestFinished(httpContext, context.StartTick, currentTick);
+            var currentTimestamp = context.StartTimestamp != 0 ? Stopwatch.GetTimestamp() : 0;
+            _logger.RequestFinished(httpContext, context.StartTimestamp, currentTimestamp);
 
             if (exception == null)
             {
                 if (_diagnosticSource.IsEnabled("Microsoft.AspNet.Hosting.EndRequest"))
                 {
-                    _diagnosticSource.Write("Microsoft.AspNet.Hosting.EndRequest", new { httpContext = httpContext, tickCount = currentTick });
+                    _diagnosticSource.Write("Microsoft.AspNet.Hosting.EndRequest", new { httpContext = httpContext, tickCount = currentTimestamp });
```

Comment:
Should this be a `TimeSpan`? For starters `tickCount` is a bit of a misnomer with this change and doesn't convey the unit of measurement. 

Also, changing this from int -> long \ TimeSpan, this would be a breaking change I know @avanderhoorn uses it in glimpse. Are there other folks we'd need to notify about this change?